sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -952,9 +925,35 @@ export class MessageReceiver extends LinkEntity {
           connectionHost: this._context.namespace.config.host,
           delayInSeconds: 15
         };
-        if (!this.wasCloseInitiated) {
-          await retry<void>(config);
-        }
+        await retry<void>(config);
+      } else if (connectionDidDisconnect && !this.wasCloseInitiated) {
+        const state: any = {
+          wasCloseInitiated: this.wasCloseInitiated,
+          receiverError: receiverError,
+          _receiver: this._receiver
+        };
+        log.error(
+          "[%s] Something went wrong and the connection disconnected. State of Receiver '%s' with address '%s' is: %O",
+          connectionId,
+          this.name,
+          this.address,
+          state
+        );
+        // Not retrying, throw the error so it gets logged and forwarded to user's error handler.
+        throw receiverError;
```

Comment:
At this point, the error is not translated right?
We should probably translate it upfront in this method

Reply:
Good catch, I've updated this to throw the translated error.